### PR TITLE
Check to be sure that the test result file has actual results before uploading

### PR DESCRIPTION
**Pull request by JamesWTruher**
# PR Summary

VSTS produces a warning if the result set is 0, we will skip uploading this to avoid the warning and better support automation scenarios.

## PR Context

We have automation which checks our test results and if the result set is 0 for that file, a warning is produced. This PR will avoid uploading that file into the test results (which is valid, as there are no run tests). We will still add the file to our artifact list for completeness.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: build.psm1 [hunk 1]
```diff
@@ -1230,7 +1230,13 @@ function Publish-TestResults
             $_ -replace 'result="Ignored"', 'result="Skipped"'
         } | Out-File -FilePath $tempFilePath -Encoding ascii -Force
 
-        Write-Host "##vso[results.publish type=$Type;mergeResults=true;runTitle=$Title;publishRunAttachments=true;resultFiles=$tempFilePath;]"
+        # If we attempt to upload a result file which has no test cases in it, then vsts will produce a warning
+        # so check to be sure we actually have a result file that contains test cases to upload.
+        # If the the "test-case" count is greater than 0, then we have results.
+        # Regardless, we want to upload this as an artifact, so this logic doesn't pertain to that.
+        if ( @(([xml](Get-Content $Path)).SelectNodes(".//test-case")).Count -gt 0 ) {
```
Comment (TravisEz13) [inline build.psm1]: The test is filtering all XUnit files...
```suggestion
        if ( @(([xml](Get-Content $Path)).SelectNodes(".//test-case") -or $Type -eq 'XUnit').Count -gt 0 ) {
```

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
Make sure XUnit results are published.

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
@PoshChan 	Please remind me in 1 hour

**Comment by PoshChan** [comment]
@TravisEz13, all requests start with the magic word: Please

**Comment by TravisEz13** [comment]
@PoshChan Please remind me in 1 hour

**Comment by PoshChan** [comment]
@TravisEz13, will remind you in 1 hour

**Comment by PoshChan** [comment]
@TravisEz13, this is the reminder you requested 1 hour ago

**Pull request merged**